projects/netgrif-components-core/src/lib/resources/engine-endpoint/task-resource.service.ts

```diff
@@ -204,41 +204,45 @@ export class TaskResourceService extends AbstractResourceService implements Coun
     public getData(taskId: string): Observable<Array<DataGroup>> {
         return this.rawGetData(taskId).pipe(
             map((responseOutcome: EventOutcomeMessageResource) => {
-                const dataGroupsArray = this.changeType((responseOutcome.outcome as GetDataGroupsEventOutcome).data, 'dataGroups');
-                if (!Array.isArray(dataGroupsArray)) {
-                    return [];
-                }
-                const result = [];
-                dataGroupsArray.forEach(dataGroupResource => {
-                    const dataFields: Array<DataField<any>> = [];
-                    if (!dataGroupResource.fields._embedded) {
-                        return; // continue
+                if (responseOutcome.success) {
```

Comment:
the condition should be inverted and only the `else` block should be a part of the if. The rest of the code then remains more readable as it has less indentation

Reply:
Done